File: mobile/concord/lib/api/models/two_factor_setup_response.dart
import 'json_utils.dart';

class TwoFactorSetupResponse {
  const TwoFactorSetupResponse({required this.secretKey, required this.otpAuthUri, required this.qrCodeSvg});

  factory TwoFactorSetupResponse.fromJson(Map<String, dynamic> json) {
    return TwoFactorSetupResponse(
      secretKey: json.field('SecretKey') as String?,
      otpAuthUri: json.field('OtpAuthUri') as String?,
      qrCodeSvg: json.field('QrCodeSvg') as String?,
    );
  }

  final String? secretKey;

  final String? otpAuthUri;

  final String? qrCodeSvg;
}

File: backend/Concord/Concord.Application/Models/TwoFactorSetupResponse.cs
using System.Text.Json.Serialization;

namespace Concord.Application.Models;

public class TwoFactorSetupResponse
{
    [JsonPropertyName("SecretKey")]
    public string? SecretKey { get; set; }

    [JsonPropertyName("OtpAuthUri")]
    public string? OtpAuthUri { get; set; }

    [JsonPropertyName("QrCodeSvg")]
    public string? QrCodeSvg { get; set; }
}
